src/group_theory/order_of_element.lean

```diff
@@ -38,6 +39,92 @@ universe u
 
 variables {α : Type u} {s : set α} {a a₁ a₂ b c: α}
 
+section is_of_fin_order
+
+variables {α} [monoid α]
+variables {H : Type u} [add_monoid H]
+
+-- Move the following four lemmas somewhere else?
+@[simp] lemma add_right_fpow_apply (x : H) {n : ℕ} : (+ x)^[n] = (+ n •ℕ x) :=
+begin
+  induction n with d hd,
+  { simp_rw [zero_nsmul, add_zero, iterate_zero],
+    refl },
+  { simp_rw [iterate_succ', succ_nsmul', ← add_assoc, hd] }
+end
+
+@[simp, to_additive] lemma mul_right_fpow_apply (a : α) {n : ℕ} : (* a)^[n] = (* a ^ n) :=
+begin
+  induction n with d hd,
+  { simp_rw [pow_zero, mul_one, iterate_zero],
+    refl },
+  { simp_rw [iterate_succ', pow_succ', ← mul_assoc, hd] }
+end
+
+@[simp] lemma add_fpow_apply_zero (x : H) {n : ℕ} : (+ x)^[n] 0 = n •ℕ x :=
+by simp
+
+@[simp, to_additive] lemma mul_fpow_apply_one (a : α) {n : ℕ} : (* a)^[n] 1 = a ^ n :=
+by simp
+
+lemma is_periodic_pt_add_iff_nsmul_eq_zero (x : H) {n : ℕ} :
+  is_periodic_pt (+ x) n 0 ↔ n •ℕ x = 0 :=
+by rw [is_periodic_pt, is_fixed_pt, add_fpow_apply_zero]
+
+@[to_additive is_periodic_pt_add_iff_nsmul_eq_zero]
+lemma is_periodic_pt_mul_iff_pow_eq_one (a : α) {n : ℕ} : is_periodic_pt (* a) n 1 ↔ a ^ n = 1 :=
+by rw [is_periodic_pt, is_fixed_pt, mul_fpow_apply_one]
+
+/-- `is_of_fin_add_order` is a predicate on an element `x` of an additive monoid to be of finite
+order, i.e. there exists `n ≥ 1` such that `n •ℕ x = 0`.-/
+def is_of_fin_add_order (x : H) : Prop :=
+(0 : H) ∈ periodic_pts (+ x)
+
+/-- `is_of_fin_order` is a predicate on an element `a` of a monoid to be of finite order, i.e. there
+exists `n ≥ 1` such that `a ^ n = 1`.-/
+@[to_additive is_of_fin_add_order]
+def is_of_fin_order (a : α) : Prop :=
+(1 : α) ∈ periodic_pts (* a)
+
+lemma foo {a : α} :
+is_of_fin_add_order (additive.of_mul a) ↔ is_of_fin_order a :=
+by refl
+
+lemma foo' {x : H} :
+is_of_fin_order (multiplicative.of_add x) ↔ is_of_fin_add_order x :=
+by refl
+
+lemma is_of_fin_add_order_iff_nsmul_eq_zero (x : H) :
+  is_of_fin_add_order x ↔ ∃ n, 0 < n ∧ n •ℕ x = 0 :=
+begin
+  split,
+  { intro h,
+    rcases h with ⟨n, hnpos, hper⟩,
+    refine ⟨n, hnpos, _⟩,
+    rwa [is_periodic_pt, is_fixed_pt, add_fpow_apply_zero] at hper },
+  { intro h,
+    rcases h with ⟨n, hnpos, hpow⟩,
+    refine ⟨n, hnpos, _⟩,
+    rwa [is_periodic_pt, is_fixed_pt, add_fpow_apply_zero] },
+end
```

Comment:
```suggestion
by { convert iff.rfl, simp only [exists_prop, is_periodic_pt_add_iff_nsmul_eq_zero] }
```

Again, this should work!

Reply:
Ah, yes. Thanks. The reason is that this lemma is older than `is_periodic_pt_add_iff_nsmul_eq_zero`.